### PR TITLE
Core/Packets: Adjust 'TalentsInspecting' option

### PR DESCRIPTION
**Pull request by Treeston**
Core/Packets: Adjust 'TalentsInspecting' option so it no longer doesn't actually do what it says (but would instead disable talent inspecting completely).

### CODE REVIEW

**Comment by tkrokli** [comment]
Looks nice, thanks for the contribution! :+1:

**Comment by DDuarte** [comment]
I think 0 and 1 should be swapped to maintain backwards compatibility. Someone that updates the core but doesn't update the config (quite common) will suddenly have inspection enabled on all characters.

**Comment by Treeston** [comment]
Which is the same behavior that setting `1` had before this change.
The only value that changes behavior is `0` (nobody can inspect -> only own faction can inspect).

**Comment by DDuarte** [comment]
In that case why isn't the default value the same?

**Comment by Treeston** [comment]
Because the new default (`0`) is blizzlike behavior.

**Comment by Warpten** [comment]
Easier to manage if you set 0,1,2 as config values then do -1

**Comment by Treeston** [comment]
Yes, but not doable while maintaining both sensible valuing and backwards compatibility.

File: src/server/game/Handlers/MiscHandler.cpp [hunk 1]
```diff
@@ -1203,7 +1203,8 @@ void WorldSession::HandleInspectOpcode(WorldPacket& recvData)
     WorldPacket data(SMSG_INSPECT_TALENT, guid_size+4+talent_points);
     data << player->GetPackGUID();
 
-    if (sWorld->getBoolConfig(CONFIG_TALENTS_INSPECTING) || _player->IsGameMaster())
+    // config value can be 0, 1 or INT_MAX - +1 guarantees that we end up with 0-3 (2-3 valid)
+    if (GetPlayer()->CanBeGameMaster() || sWorld->getIntConfig(CONFIG_TALENTS_INSPECTING) + (GetPlayer()->GetTeamId() == player->GetTeamId())+1 > 1)
```
Comment (jackpoz) [inline src/server/game/Handlers/MiscHandler.cpp]: do we really need arithmetic operations on boolean type ?

File: src/server/game/Handlers/MiscHandler.cpp [hunk 2]
```diff
@@ -1203,7 +1203,8 @@ void WorldSession::HandleInspectOpcode(WorldPacket& recvData)
     WorldPacket data(SMSG_INSPECT_TALENT, guid_size+4+talent_points);
     data << player->GetPackGUID();
 
-    if (sWorld->getBoolConfig(CONFIG_TALENTS_INSPECTING) || _player->IsGameMaster())
+    // config value can be 0, 1 or INT_MAX - +1 guarantees that we end up with 0-3 (2-3 valid)
```
Comment (jackpoz) [inline src/server/game/Handlers/MiscHandler.cpp]: is INT_MAX actually a valid config value ?

Comment (Treeston) [inline src/server/game/Handlers/MiscHandler.cpp]: `getIntConfig` returns `uint32`, `-1` -> `INT_MAX`, unsigned overflow is well-defined.

Comment (Shauren) [inline src/server/game/Handlers/MiscHandler.cpp]: -1 is not INT_MAX. (INT_MAX is 0x7FFFFFFF, or 2147483647; INT_MIN is 0x80000000, or -2147483648)

Comment (Treeston) [inline src/server/game/Handlers/MiscHandler.cpp]: Whatever the max value of uint32 is then (2*INT_MAX +-1 or something). Doesn't really make a difference in functionality, I can change the comment.

Comment (Shauren) [inline src/server/game/Handlers/MiscHandler.cpp]: then change the comment?

**Pull request merged**